components/rhs_thread/rhs_thread.tsx

```diff
@@ -32,40 +35,53 @@ export function renderThumbHorizontal() {
     return (<div/>);
 }
 
-export function renderThumbVertical(props) {
+export function renderThumbVertical(props: Props) {
     return (
         <div
             {...props}
             className='scrollbar--vertical'
         />);
 }
 
-export default class RhsThread extends React.Component {
-    static propTypes = {
-        posts: PropTypes.arrayOf(PropTypes.object).isRequired,
-        channel: PropTypes.object.isRequired,
-        selected: PropTypes.object.isRequired,
-        previousRhsState: PropTypes.string,
-        currentUserId: PropTypes.string.isRequired,
-        previewCollapsed: PropTypes.string.isRequired,
-        previewEnabled: PropTypes.bool.isRequired,
-        socketConnectionStatus: PropTypes.bool.isRequired,
-        actions: PropTypes.shape({
-            removePost: PropTypes.func.isRequired,
-            selectPostCard: PropTypes.func.isRequired,
-            getPostThread: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Props = {
+    posts: Post[];
+    channel: Partial<Channel> | null;
+    selected: Record<string, any>;
+    previousRhsState?: string;
+    currentUserId: string;
+    previewCollapsed: string;
+    previewEnabled: boolean;
+    socketConnectionStatus: boolean;
+    actions: {
+        removePost: (post: ExtendedPost) => (dispatch: DispatchFunc, getState: GetStateFunc) => void;
+        selectPostCard: (post: Record<string, any>) => Record<string, any>;
+        getPostThread: (rootId: string, root?: boolean) => void;
+    };
+}
+
+type State = {
+    selected?: Record<string, any>;
+    windowWidth?: number;
+    windowHeight?: number;
+    isScrolling: boolean;
+    topRhsPostId: number | string;
+    openTime: number;
+    postsArray?: Record<string, any>[];
+    isBusy?: boolean;
+}
+
+export default class RhsThread extends React.Component<Props, State> {
+    private scrollStopAction: DelayedAction;
 
-    static getDerivedStateFromProps(props, state) {
-        let updatedState = {selected: props.selected};
+    public static getDerivedStateFromProps(props: Props, state: State) {
+        let updatedState: Record<string, any> = {selected: props.selected};
```

Comment:
Perhaps we could use a `Partial` here?
```suggestion
        let updatedState: Partial<State> = {selected: props.selected};
```